core/services/vrf/v2/listener_v2_log_listener.go

```diff
@@ -39,36 +40,35 @@ func (lsn *listenerV2) runLogListener(
 		case <-ticker.C:
 			start := time.Now()
 			lsn.l.Debugw("log listener loop")
-			// Filter registration is idempotent, so we can just call it every time
-			// and retry on errors using the ticker.
-			err := lsn.chain.LogPoller().RegisterFilter(logpoller.Filter{
-				Name: logpoller.FilterName(
-					"VRFListener",
-					"version", lsn.coordinator.Version(),
-					"keyhash", lsn.job.VRFSpec.PublicKey.MustHash(),
-					"coordinatorAddress", lsn.coordinator.Address()),
-				EventSigs: evmtypes.HashArray{
-					lsn.coordinator.RandomWordsFulfilledTopic(),
-					lsn.coordinator.RandomWordsRequestedTopic(),
-				},
-				Addresses: evmtypes.AddressArray{
-					lsn.coordinator.Address(),
-				},
-			})
-			if err != nil {
-				lsn.l.Errorw("error registering filter in log poller, retrying",
-					"err", err,
-					"elapsed", time.Since(start))
-				continue
+
+			// If filter has not already been successfully registered, register it.
+			if !lsn.chain.LogPoller().HasFilter(filterName) {
+				err := lsn.chain.LogPoller().RegisterFilter(logpoller.Filter{
+					Name: filterName,
+					EventSigs: evmtypes.HashArray{
+						lsn.coordinator.RandomWordsFulfilledTopic(),
+						lsn.coordinator.RandomWordsRequestedTopic(),
+					},
+					Addresses: evmtypes.AddressArray{
+						lsn.coordinator.Address(),
+					},
+				})
+				if err != nil {
+					lsn.l.Errorw("error registering filter in log poller, retrying",
+						"err", err,
+						"elapsed", time.Since(start))
+					continue
+				}
 			}
 
 			// on startup we want to initialize the last processed block
 			if startingUp {
+				var err2 error
```

Comment:
I think it should be fine naming it as `err` instead of `err2`?

Reply:
Yeah, I think you're right that it wouldn't cause a shadowing issue.